prime-router/docs/design/proposals/0023-condition-to-code-mapping/0023-condition-to-code-mapping.md

```diff
@@ -0,0 +1,229 @@
+
+### Filtering for condition
+
+Due to the presence of the tag on the fhir bundle we can utilize any of the existing filters to manage the list of conditions for each receiver. There are several potential options for filtering.
+
+1.) Jurisdictional filter - This option would remove visibility into which messages qualified for the STLT but were filtered out due to condition it is not recommended to use the jurisdictional filter for this purpose.<br>
+2.) Condition filter - This option allows visibility into which messages were filtered out due to condition, and visually separates the condition logic from the other filters. This is the proposed method.
+
+There are multiple ways to write the logic in the condition filter. The first way will remove any observations that do not pass the condition filter.<br>
+
+### Condition filter logic #1
+
+Example input: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleinput.fhir]()
+
+Example condition logic:
+```yaml
+    - name: TEST-RECEIVER
+      externalName: TEST
+      organizationName: TEST
+      topic: full-elr
+      customerStatus: active
+      jurisdictionalFilter:
+        - "(%performerState.exists() and %performerState = 'TEST') or (%patientState.exists() and %patientState = 'TEST')"
+      qualityFilter: []
+      conditionFilter:
+          - "%resource.where(meta.tag.code in (398102009' |'13906002' |'409498004' |'40610006' |'21061004' |'75702008' |'359761005' |'414015000' |'840539006' |'416925005' |'83436008')).exists()"
+      timing:
+        operation: MERGE
+        numberPerDay: 1440 # Every minute
+        initialTime: 00:00
+```
+Example output: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleoutput1.hl7]()
+
+This will result in a message with potentially missing results in the event that there is a missing mapping, and also presents difficulties in dealing with culture results as it would require all the LOINCS for all possible micro cultures to be mapped.
+
+### Condition filter logic #2
+
+The second way to write the condition logic will pass the entire bundle if a match is found for one observation.
+
+Example input: [prime-reportstream/prime-router/docs/design/proposals/0023-condition-to-code-mapping/exampleinput.fhir]()
+
+```yaml
+    - name: TEST-RECEIVER
+      externalName: TEST
+      organizationName: TEST
+      topic: full-elr
+      customerStatus: active
+      jurisdictionalFilter:
+        - "(%performerState.exists() and %performerState = 'TEST') or (%patientState.exists() and %patientState = 'TEST')"
+      qualityFilter: []
+      conditionFilter:
+          - "bundle.entry.ofType(Observation).where(meta.tag.code in (398102009' |'13906002' |'409498004' |'40610006' |'21061004' |'75702008' |'359761005' |'414015000' |'840539006' |'416925005' |'83436008')).exists()"
```

Comment:
Is there another way we can achieve this without having to have the codes in our filters? Could we query RCKMS to see what conditions we need to let through for each jurisdiction?

Reply:
No unfortunately. Long term that's the hope, but currently RCKMS doesn't have this functionality.

Reply:
We could have a valueset/lookuptable for each jurisdiction and have a custom fhir function that checks to see if the value is in that valueset, although that is essentially the same thing, just hiding the codes from the filters and then we couldn't change the conditions without a PR